core/store/migrate/migrations/0158_update_job_proposal_spec_job_proposal_statuses.sql

```diff
@@ -0,0 +1,144 @@
+-- +goose Up
+-- +goose StatementBegin
+-- JobProposals Table
+-- We must remove the old contraint to add an enum value to support Postgres v11
+ALTER TABLE job_proposals DROP CONSTRAINT chk_job_proposals_status_fsm;
+
+-- Drop the cancelled enum value. Unfortunately postgres does not support a
+-- a way to remove a value from an enum.
+ALTER TYPE job_proposal_status
+RENAME TO job_proposal_status_old;
+
+CREATE TYPE job_proposal_status AS ENUM(
+	'pending',
+	'approved',
+	'rejected',
+	'cancelled',
+	'deleted',
+	'revoked'
+);
+
+ALTER TABLE job_proposals
+ALTER COLUMN STATUS TYPE job_proposal_status USING STATUS::text::job_proposal_status;
+
+DROP TYPE job_proposal_status_old;
+
+ALTER TABLE job_proposals
+ADD CONSTRAINT chk_job_proposals_status_fsm CHECK (
+		(
+			STATUS = 'pending'
+			AND external_job_id IS NULL
+		)
+		OR (
+			STATUS = 'approved'
+			AND external_job_id IS NOT NULL
+		)
+		OR (
+			STATUS = 'rejected'
+			AND external_job_id IS NULL
+		)
+		OR (
+			STATUS = 'cancelled'
+			AND external_job_id IS NULL
+		)
+		OR (
+			STATUS = 'deleted'
+			AND external_job_id IS NULL
+		)
+		OR (
+			STATUS = 'revoked'
+			AND external_job_id IS NULL
+		)
+	);
+
+-- JobProposalSpecs Table
+-- We must remove the old constraint to add an enum value to support Postgres v11
+DROP INDEX idx_job_proposal_specs_job_proposal_id_and_status;
+
+-- Drop the cancelled enum value. Unfortunately postgres does not support a
+-- a way to remove a value from an enum.
+ALTER TYPE job_proposal_spec_status
```

Comment:
You shouldn't need to add any new spec status

Reply:
was kind of going back and forth with this... yeah, I think cancelling them is fine, altho the `CancelSpec` orm method also cancels the proposal. I guess it's probably fine leaving them as approved if the proposal itself is deleted.